src/main/java/org/truffleruby/Layouts.java

```diff
@@ -101,6 +101,14 @@
 
 public abstract class Layouts {
 
+    // Standard identifiers
+    // These must appear before the generated layout list so the identifiers have been initialized by the time
+    // the layout singletons are created.
+
+    public static final HiddenKey OBJECT_ID_IDENTIFIER = new HiddenKey("object_id");
```

Comment:
35e1746a9beb65c7141b175e0bb487a0678574c8 fixes the case of calling object_id on Strings in IO#puts.
I think object_id is different than tainted/frozen which are mostly String-specific.
If we add object_id we might as well do it for all objects, which is a different decision.